hw/xbox/nv2a/nv2a_psh.c

```diff
@@ -295,24 +296,24 @@ static QString* get_input_var(struct PixelShader *ps, struct InputInfo in, bool
 {
     QString *reg = get_var(ps, in.reg, false);
 
-    if (strcmp(qstring_get_str(reg), "0.0") != 0
-        && (in.reg != PS_REGISTER_EF_PROD
-            || strstr(qstring_get_str(reg), ".a") == NULL)) {
-        switch (in.chan) {
-        case PS_CHANNEL_RGB:
-            if (is_alpha) {
-                qstring_append(reg, ".b");
-            } else {
-                qstring_append(reg, ".rgb");
-            }
-            break;
-        case PS_CHANNEL_ALPHA:
+    switch (in.chan) {
+    case PS_CHANNEL_RGB:
+        if (is_alpha) {
+            qstring_append(reg, ".b");
+        } else {
+            qstring_append(reg, ".rgb");
+        }
+        break;
+    case PS_CHANNEL_ALPHA:
+        if (is_alpha) {
             qstring_append(reg, ".a");
-            break;
-        default:
-            assert(false);
-            break;
+        } else {
+            qstring_append(reg, ".aaa");
```

Comment:
Makes sense:
```
     while an ALPHA <componentUsage> parameter indicates that the
     alpha portion of the indicated register should be replicated across
     the RGB portion of the combiner input variable.
```

Reply:
The old code also did this, but using ugly `vec3` casts when accessing the values (which still exist, and will be cleaned up in the future).

See #202

Reply:
Agree, the old `vec3` casting approach was ugly. This method is much more readable and clearly indicates desired action/intent at this point. Thanks!